cpp/src/io/parquet/page_data.cu

```diff
@@ -815,6 +815,37 @@ inline __device__ void gpuOutputDecimalAsFloat(volatile page_state_s *s,
   *dst  = (scale < 0) ? (d * kPow10[min(-scale, 39)]) : (d / kPow10[min(scale, 39)]);
 }
 
+/**
+ * @brief Output a fixed-length byte array(len <= 8) as a 64-bit int
+ *
+ * @param[in,out] s Page state input/output
+ * @param[in] src_pos Source position
+ * @param[in] dst Pointer to row output data
+ */
+inline __device__ void gpuOutputFixedLenByteArrayAsInt64(volatile page_state_s *s,
+                                                         int src_pos,
+                                                         int64_t *dst)
+{
+  uint32_t const dtype_len_in = s->dtype_len_in;
+  uint8_t const *dict         = s->dict_base ? s->dict_base : s->data_start;
+  uint32_t const dict_pos =
+    (s->dict_base ? ((s->dict_bits > 0) ? s->dict_idx[src_pos & (non_zero_buffer_size - 1)] : 0)
+                  : src_pos) *
+    dtype_len_in;
+  uint32_t const dict_size = s->dict_size;
+
+  int64_t unscaled64 = 0;
+  for (unsigned int i = 0; i < dtype_len_in; i++) {
+    uint32_t v = (dict_pos + i < dict_size) ? dict[dict_pos + i] : 0;
+    unscaled64 = (unscaled64 << 8) | v;
+  }
+  if (dtype_len_in < 8) {
+    unscaled64 <<= 64 - dtype_len_in * 8;
```

Comment:
We should comment this a little. Something like // If the value isn't all 8 bytes we need to shift it up and back down in order to sign extend the value so a negative number is correctly represented. Someone who doesn't know this trick will appreciate the comment.

Reply:
Good point! It's done.